packetbeat/protos/drda/drda.go

```diff
@@ -0,0 +1,562 @@
+	}
+
+	trans.Requests[fmt.Sprint(drdaAbbrev(msg.ddm.Codepoint), "_", msg.ddm.Cor)] = common.MapStr{
+
+		"description":  drda_description[msg.ddm.Codepoint],
+		"codepoint":    msg.ddm.Codepoint,
+		"length2":      msg.ddm.Length2,
+		"format":       msg.ddm.Format,
+		"format_flags": msg.ddm.DSSFlags,
+		//"format_reserved":        msg.ddm.DSSFlags >> DRDA_DSSFMT_RESERVED,
+		//"format_chained":         msg.ddm.DSSFlags >> DRDA_DSSFMT_CHAINED,
+		//"format_continue":        msg.ddm.DSSFlags >> DRDA_DSSFMT_CONTINUE,
+		//"format_samecor":         msg.ddm.DSSFlags >> DRDA_DSSFMT_SAME_CORR,
+		"format_dss_type":        msg.ddm.DSSType,
+		"format_dss_type_abbrev": dss_abbrev[uint16(msg.ddm.DSSType)],
+		"correlation_id":         msg.ddm.Cor,
+		"length":                 msg.ddm.Length,
+		"direction":              msg.Direction,
+		"parameters":             tmp,
+	}
+
+	trans.Notes = msg.Notes
+	trans.BytesIn += uint64(msg.ddm.Length)
+}
+
+func (drda *Drda) receivedDrdaResponse(msg *DrdaMessage) {
+	trans := drda.getTransaction(msg.TcpTuple.Hashable())
+
+	if trans == nil {
+		logp.Err("No transaction for this response")
+	}
+
+	if trans.Responses == nil {
+		trans.Responses = common.MapStr{}
+	}
+
+	tmp := common.MapStr{}
+
+	for key, value := range msg.parameters {
+
+		p := common.MapStr{}
+		p["desc"] = drda_description[key]
+		p["codepoint"] = value.Codepoint
+		p["length"] = value.Length
+		p["data_ascii"] = value.ASCIIData
+		p["data_ebcdic"] = value.EBCDICData
+
+		tmp[drdaAbbrev(key)] = p
+	}
+
+	trans.Responses[fmt.Sprint(drdaAbbrev(msg.ddm.Codepoint), "_", msg.ddm.Cor)] = common.MapStr{
+
+		"description":  drda_description[msg.ddm.Codepoint],
+		"codepoint":    msg.ddm.Codepoint,
+		"length2":      msg.ddm.Length2,
+		"format":       msg.ddm.Format,
+		"format_flags": msg.ddm.DSSFlags,
+		//"format_reserved":        msg.ddm.DSSFlags >> DRDA_DSSFMT_RESERVED,
+		//"format_chained":         msg.ddm.DSSFlags >> DRDA_DSSFMT_CHAINED,
+		//"format_continue":        msg.ddm.DSSFlags >> DRDA_DSSFMT_CONTINUE,
+		//"format_samecor":         msg.ddm.DSSFlags >> DRDA_DSSFMT_SAME_CORR,
+		"format_dss_type":        msg.ddm.DSSType,
+		"format_dss_type_abbrev": dss_abbrev[uint16(msg.ddm.DSSType)],
+		"correlation_id":         msg.ddm.Cor,
+		"length":                 msg.ddm.Length,
+		//"direction": msg.Direction,
+		"parameters": tmp,
+	}
+
+	trans.BytesOut += uint64(msg.ddm.Length)
+	trans.Notes = append(trans.Notes, msg.Notes...)
+}
+
+func (drda *Drda) publishTransaction(t *DrdaTransaction) {
+
+	if drda.results == nil {
+		logp.Err("Nothing to publish")
+		return
+	}
+
+	t.ResponseTime = int32(t.TsEnd.Sub(t.TsStart).Nanoseconds() / 1e6) // resp_time in milliseconds
+
+	event := common.MapStr{}
+	event["type"] = "drda"
+
+	event["responsetime"] = t.ResponseTime
+	if drda.Send_request {
+		event["request"] = "n.a." //t.Request_raw
+	}
+	if drda.Send_response {
+		event["response"] = "n.a." //t.Response_raw
+	}
+
+	event["query"] = t.Query
```

Comment:
I see now that some code for the above comment exists, but it doesn't seem to work?